src/Explorer/Web/ContractView.hs

```diff
@@ -208,10 +217,59 @@ renderCTVRs ctvrs = table ! style "border: 1px solid black" $ do
             td $ toHtml . ctvrSlot $ ctvr
     forM_ ctvrs makeRow
 
+renderParty :: Party -> String
+renderParty (Address ad) = printf "Address: %s" $ unpack ad
+renderParty (Role ro) = printf "Role: %s" $ unpack ro
+
+renderToken :: Token -> String
+renderToken (Token "" "") = "ADA (Lovelace)"
+renderToken (Token currSymbol tokenName) = printf "%s (%s)" currSymbol tokenName
+
+renderMAccounts :: Map (Party, Token) Money -> Html
+renderMAccounts mapAccounts = table ! style "border: 1px solid black" $ do
+  tr $ do
+    th $ b "party"
+    th $ b "currency (token name)"
+    th $ b "amount"
+  let mkRow ((party, token), money) =
```

Comment:
Why not `where`

Reply:
eh, for some reason I always use `let` in do blocks but it's all the same, just personal preference